rs/ic_os/config/src/types.rs

```diff
@@ -0,0 +1,125 @@
+use ic_types::malicious_behaviour::MaliciousBehaviour;
+use serde::{Deserialize, Serialize};
+use std::net::{Ipv4Addr, Ipv6Addr};
+use std::path::PathBuf;
+use url::Url;
+
+/// SetupOS configuration. User-facing configuration files
+/// (e.g., `config.ini`, `deployment.json`) are transformed into `SetupOSConfig`.
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+pub struct SetupOSConfig {
+    pub network_settings: NetworkSettings,
+    pub icos_settings: ICOSSettings,
+    pub setupos_settings: SetupOSSettings,
+    pub hostos_settings: HostOSSettings,
+    pub guestos_settings: GuestOSSettings,
+}
+
+/// HostOS configuration. In production, this struct inherits settings from `SetupOSConfig`.
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+pub struct HostOSConfig {
+    pub network_settings: NetworkSettings,
+    pub icos_settings: ICOSSettings,
+    pub hostos_settings: HostOSSettings,
+    pub guestos_settings: GuestOSSettings,
+}
+
+/// GuestOS configuration. In production, this struct inherits settings from `HostOSConfig`.
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+pub struct GuestOSConfig {
+    pub network_settings: NetworkSettings,
+    pub icos_settings: ICOSSettings,
+    pub guestos_settings: GuestOSSettings,
+}
+
+/// Placeholder for SetupOS-specific settings.
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+pub struct SetupOSSettings;
+
+/// HostOS-specific settings.
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+pub struct HostOSSettings {
+    pub vm_memory: u32,
+    pub vm_cpu: String,
+    pub verbose: bool,
+}
+
+/// GuestOS-specific settings.
+#[derive(Serialize, Deserialize, Debug, PartialEq, Default, Clone)]
+pub struct GuestOSSettings {
+    /// Externally generated cryptographic keys.
+    /// Must be a directory with contents matching the internal representation of the ic_crypto directory.
+    /// When given, this provides the private keys of the node.
+    /// If not given, the node will generate its own private/public key pair.
+    pub ic_crypto_path: Option<PathBuf>,
+    pub ic_state_path: Option<PathBuf>,
+    /// Initial registry state.
+    /// Must be a directory with contents matching the internal representation of the ic_registry_local_store.
+    /// When given, this provides the initial state of the registry.
+    /// If not given, the node will fetch (initial) registry state from the NNS.
```

Comment:
I moved a lot of configuration documentation from guestos/docs/ConfigStore.adoc to here. WDYT?